hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/BaseActionExecutor.java

```diff
@@ -58,7 +59,7 @@ public BaseActionExecutor(HoodieEngineContext context, HoodieWriteConfig config,
    */
   protected final void writeTableMetadata(HoodieCommitMetadata metadata, String actionType) {
     table.getMetadataWriter(instantTime).ifPresent(w -> w.update(
-        metadata, instantTime, table.isTableServiceAction(actionType)));
+        metadata, instantTime, table.isTableServiceAction(new HoodieInstant(HoodieInstant.State.INFLIGHT, actionType, instantTime))));
```

Comment:
not a big fan of creating dummy instant to cater the API. it's the API that needs to fit better with the caller info.

Reply:
we need entire HoodieInstant to get the clstuering plan. so not sure what other option we have. If you have any solution, let me know. happy to incorporate.

Reply:
@nsivabalan seeing there is `org.apache.hudi.common.table.timeline.HoodieInstant.State#INVALID` which is only used by tests, so we can change it to something like `NIL` and use it for this scenario? At least to make it clear that this is a dummy instant, better than marking it inflight.